docs/01-basics/01-basic-window.md

```diff
@@ -0,0 +1,286 @@
+  when we have a render context available. What that is I will explain later.
+  ```cmake title="lib/CMakeLists.txt"
+  --8<-- "lib/CMakeLists.txt:24:42"
+  ```
+
+`spdlog`
+
+: a logging framework which provides structured logging facilities. No more weird `printf` or `std::cout`.
+  ```cmake title="lib/CMakeLists.txt"
+  --8<-- "lib/CMakeLists.txt:44:55"
+  ```
+
+Next one is `CMakeLists.txt` in our project directory.
+
+That is our main project file.
+
+```cmake title="CMakeLists.txt"
+--8<-- "CMakeLists.txt:1:7"
+```
+
+It declares the project name, which c++ standard we want to target and which targets we want to add.
+Those are a) our dependencies which are declared in `lib/CMakeLists.txt` hence the `add_directory(lib)` line
+as well as b) our actual projects we are going to write in this guide.
+
+Let's put in our first project in too, right below `add_subdirectory(lib)`
+
+```cmake
+add_subdirectory(src/00-BasicWindow)
+```
+
+And then we move on to BasicWindow's CMakeLists.txt and Main.cpp
+
+```cmake title="src/01-01-BasicWindow/CMakeLists.txt"
+--8<-- "src/01-01-BasicWindow/CMakeLists.txt"
+```
+
+Again declare a target, in this case `01-01-BasicWindow` (I try to keep the docs pages in sync with the target names to aid my brain while writing this).
+Specify which cpp file/unit to build.
+
+Then there is a small block which detects whether you are using Microsoft's C++ Compiler or not, and
+configure the compiler to always treat warnings as errors.
+
+Then we make `spdlog`'s include directory available to the target and link the program with our dependencies.
+
+Finally we can start working on the first c++ parts. We will keep everything in one file, to create a window and react to keyboard input `ESC` and `F11`.
+
+Let's start.
+
+### Main.cpp
+
+I like to use `int32_t`, `uint16_t` over types/aliases like `int`, or `ushort`. For that we include `cstdint`.
+
+We also include glad and glfw header as well as spdlog's. When it comes to glad and glfw we need to be careful here to include glad before glfw.
+C++ is weird and sometimes the order of includes is important, because certain compiler definitions/switches are declared somewhere else which are picked up by something else.
+
+That's one of the cases here.
+
+Plus the usual entry point for your bog standard c/c++ program.
+
+It all should look like this:
+
+```cpp
+#include <cstdint>
+
+#include <glad/glad.h>
+#include <GLFW/glfw3.h>
+
+#include <spdlog/spdlog.h>
+
+int main(int32_t argc, char* argv[])
+{
+    return 0;   
+}
+```
+
+You might want to reconfigure the project here, to be sure all deps are loaded and `CMake` knows how to build all this stuff.
+in `VSCode` press `Ctrl+Shift+p` and look for `CMake: Delete Cache and Reconfigure`. It is probably called very similar in `Clion` or `Visual Studio`.
+
+`VSCode` will probably say something like this afterwards:
```

Comment:
~~will probably say something like this afterwards~~
writes the following to its output window when reconfiguration is done